src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -179,6 +179,35 @@ public Uri Endpoint
         /// </remarks>
         public Func<HttpClient> HttpClientFactory { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum number of call attempts. This value
+        /// includes the original attempt. Used to set
+        /// <c>Grpc.Net.Client.Configuration.RetryPolicy.MaxAttempts</c>.
+        /// </summary>
+        /// <remarks>
+        /// Setting this value to less than 2 will disable retry attempts.
+        /// </remarks>
+        public int RetryMaxAttempts { get; set; } = 5;
```

Comment:
Do we want to expose these options publicly at this time? I summarized some thoughts and findings awhile back here with the opentelemetry-js project https://github.com/open-telemetry/opentelemetry-js/issues/1233.

Most notably, Java has implemented retry, but has left the configuration internal since the spec has not decided on defaults.

Reply:
Since the spec does not specify these details it would be better to keep these non-public for now. @alanwest should retries be enabled by default? or there should be some option to enable it?

Reply:
My opinion is to enable it by default. The spec for OTLP/gRPC seems reasonably clear that clients should have a retry mechanism by default. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#failures

Namely these two statements:

> The server MUST indicate retryable errors using code [Unavailable](https://godoc.org/google.golang.org/grpc/codes) and MAY supply additional [details via status](https://godoc.org/google.golang.org/grpc/status#Status.WithDetails) using [RetryInfo](https://github.com/googleapis/googleapis/blob/6a8c7914d1b79bd832b5157a09a9332e8cbd16d4/google/rpc/error_details.proto#L40) containing 0 value of RetryDelay.

and

> The server MAY use other gRPC codes to indicate retryable and not-retryable errors if those other gRPC codes are more appropriate for a particular erroneous situation. The client SHOULD interpret gRPC status codes as retryable or not-retryable according to the following table:

As for finer-grained configuration of the retry policy, I think we leave that TBD until and if the spec adopts configuration settings.

Reply:
Spec says "Transient errors MUST be handled with a retry strategy" https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#retry so I feel like it should be enabled by default. Maybe not enabled by default yet though..

I think it's fair to leave out the options for now and enable as users ask for more control over the retry logic.